src/safeposix/syscalls/fs_calls.rs

```diff
@@ -61,62 +101,77 @@ impl Cage {
                     );
                 }
 
-                let filename = truepath.file_name().unwrap().to_str().unwrap().to_string(); //for now we assume this is sane, but maybe this should be checked later
-
+                // Error is thrown when the input flags contain S_IFCHR flag representing a special character file.
                 if S_IFCHR == (S_IFCHR & flags) {
                     return syscall_error(Errno::EINVAL, "open", "Invalid value in flags");
                 }
 
-                let effective_mode = S_IFREG as u32 | mode;
-
+                // Check for the condition if the mode bits are correct and have the required permissions to create a directory
                 if mode & (S_IRWXA | S_FILETYPEFLAGS as u32) != mode {
                     return syscall_error(Errno::EPERM, "open", "Mode bits were not sane");
-                } //assert sane mode bits
+                }
 
+                let filename = truepath.file_name().unwrap().to_str().unwrap().to_string(); //for now we assume this is sane, but maybe this should be checked later
                 let time = interface::timestamp(); //We do a real timestamp now
+                let effective_mode = S_IFREG as u32 | mode;
```

Comment:
Why is a regular file type flag added to the effective mode flags?

Reply:
When creating a new file, we need to specify its type. S_IFREG is the flag for a regular file, so it’s added to the mode to indicate that the new file being created is a regular file.